### PR TITLE
Fix/pytest warnings

### PR DESCRIPTION
**Pull request by b-butler**
## Description
Removes pytest warnings about test functions returning values, and this was a good reminder to clean up the test code in `test/operation.py` as well. The PR uses fixtures to modify an operation through its various states for use in testing.

## Motivation and context

Prevents future errors in upcoming pytest versions.

Resolves #???

## How has this been tested?

Tests pass and semantically have not be changed except to test more.

## Change log
Not user facing.
```

```

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/CONTRIBUTING.rst).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/ContributorAgreement.md).
- [x] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

**Comment by joaander** [comment]
After merging #1551, getting these warnings now:
```
pytest/test_simulation.py::test_initialization[CPU]                                                         
  /Users/joaander/venv/lib/python3.11/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <function Simulation.__del__ at 0x1064a7ec0>                              
                                                                                                                                                                                                                          0x140f523  Traceback (most recent call last):                                                                                                                                                                                     
    File "/Users/joaander/build/hoomd/hoomd/simulation.py", line 466, in __del__                                                                                                                                         
      self._operations._unschedule()                                                                        
      ^^^^^^^^^^^^^^^^                                                                                      
  AttributeError: 'Simulation' object has no attribute '_operations'                                                                                                                                                     
                                                      
    warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))    
```
```
md/pytest/test_alchemostat.py::test_after_attaching[CPU-NVT-extra_property_1st_value0-extra_property_2nd_value0]                                                                                                         
  /Users/joaander/venv/lib/python3.11/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <function Simulation.__del__ at 0x1064a7ec0>                              
                                                                                                                                                                                                                         
  Traceback (most recent call last):                                                                                                                                                                                     
    File "/Users/joaander/build/hoomd/hoomd/simulation.py", line 466, in __del__                                                                                                                                         
      self._operations._unschedule()                                                                                                                                                                                     
    File "/Users/joaander/build/hoomd/hoomd/operations.py", line 194, in _unschedule                                                                                                                                     
      self._integrator._detach()                                                                                                                                                                                         
    File "/Users/joaander/build/hoomd/hoomd/operation.py", line 259, in _detach                                                                                                                                          
      self._detach_hook()                                                                                                                                                                                                
    File "/Users/joaander/build/hoomd/hoomd/md/integrate.py", line 60, in _detach_hook                                                                                                                                   
      self._methods._unsync()                                                                                                                                                                                            
    File "/Users/joaander/build/hoomd/hoomd/data/syncedlist.py", line 259, in _unsync                                                                                                                                    
      self._unregister_item(item)                                                                                                                                                                                        
    File "/Users/joaander/build/hoomd/hoomd/data/syncedlist.py", line 207, in _unregister_item                                                                                                                           
      value._detach()                                                                                                                                                                                                    
    File "/Users/joaander/build/hoomd/hoomd/operation.py", line 259, in _detach                                                                                                                                          
      self._detach_hook()                                                                                                                                                                                                
    File "/Users/joaander/build/hoomd/hoomd/md/alchemy/methods.py", line 36, in _detach_hook                                                                                                                             
      self._alchemical_dof._unsync()                                                                                                                                                                                     
    File "/Users/joaander/build/hoomd/hoomd/data/syncedlist.py", line 259, in _unsync                                                                                                                                    
      self._unregister_item(item)                                                                           
    File "/Users/joaander/build/hoomd/hoomd/data/syncedlist.py", line 207, in _unregister_item                                                                                                                           
      value._detach()                                                                                                                                                                                                    
    File "/Users/joaander/build/hoomd/hoomd/operation.py", line 259, in _detach                                                                                                                                          
      self._detach_hook()                             
    File "/Users/joaander/build/hoomd/hoomd/md/alchemy/pair.py", line 202, in _detach_hook                                                                                                                               
      self._force._cpp_obj.disableAlchemicalPairParticle(self._cpp_obj)                                                                                                                                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                                                                                                                 
  AttributeError: 'NoneType' object has no attribute 'disableAlchemicalPairParticle'        
```
```
md/pytest/test_active_rotational_diffusion.py::test_pickling[CPU-ActiveOnManifold]                                                                                                                                       
  /Users/joaander/venv/lib/python3.11/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <function Simulation.__del__ at 0x1064a7ec0>                              
                                                                                                                                                                                                                         
  Traceback (most recent call last):                                                                                                                                                                                     
    File "/Users/joaander/build/hoomd/hoomd/simulation.py", line 466, in __del__                                                                                                                                         
      self._operations._unschedule()                                                                                                                                                                                     
    File "/Users/joaander/build/hoomd/hoomd/operations.py", line 194, in _unschedule                                                                                                                                     
      self._integrator._detach()                                                                                                                                                                                         
    File "/Users/joaander/build/hoomd/hoomd/operation.py", line 259, in _detach                                                                                                                                          
      self._detach_hook()                                                                                                                                                                                                
    File "/Users/joaander/build/hoomd/hoomd/md/integrate.py", line 59, in _detach_hook                                                                                                                                   
      self._forces._unsync()                                                                                                                                                                                             
    File "/Users/joaander/build/hoomd/hoomd/data/syncedlist.py", line 259, in _unsync                                                                                                                                    
      self._unregister_item(item)                                                                                                                                                                                        
    File "/Users/joaander/build/hoomd/hoomd/data/syncedlist.py", line 207, in _unregister_item                                                                                                                           
      value._detach()                                                                                                                                                                                                    
    File "/Users/joaander/build/hoomd/hoomd/operation.py", line 257, in _detach                                                                                                                                          
      raise err                                                                                                                                                                                                          
    File "/Users/joaander/build/hoomd/hoomd/operation.py", line 255, in _detach                                                                                                                                          
      self._notify_disconnect()                                                                                                                                                                                          
    File "/Users/joaander/build/hoomd/hoomd/operation.py", line 157, in _notify_disconnect                                                                                                                               
      dependent._handle_removed_dependency(self)                                                                                                                                                                         
    File "/Users/joaander/build/hoomd/hoomd/md/update.py", line 295, in _handle_removed_dependency                                                                                                                       
      raise SimulationDefinitionError(                                                                                                                                                                                   
  hoomd.error.SimulationDefinitionError: The active force this updater is dependent on is being removed. Remove this updater first to avoid error.                                                                       
                                                                                                                                                                                                                         
    warnings.warn(pytest.PytestUnraisableExceptionWarning(msg)) 
```

**Review by tommy-waltmann — COMMENTED** [review]
The code looks good. Just need to fix the new warnings that are now being generated.

**Comment by b-butler** [comment]
Before merging I need to fix other warnings that I missed as my hodges dev environment had an older version of pytest.

**Comment by b-butler** [comment]
pre-commit.ci autofix

File: hoomd/simulation.py [hunk 1]
```diff
@@ -463,7 +463,11 @@ def run(self, steps, write_at_start=False):
 
     def __del__(self):
         """Clean up dangling references to simulation."""
-        self._operations._unschedule()
+        # del is called apparently when the arguments passed to init are wrong
+        # as well. Since _operations is only set at __init__ time we have to
+        # return early not to raise an exception.
```
Comment (tommy-waltmann) [inline hoomd/simulation.py]:
```suggestion
        # _operations may not be set, check before unscheduling
```

**Review by tommy-waltmann — CHANGES_REQUESTED** [review]
Just one friendly suggestion for re-wording a comment.

**Comment by joaander** [comment]
Please also merge `trunk-patch` (now 4.0.0) and resolve conflicts.

**Review by tommy-waltmann — APPROVED** [review]
LGTM

**Pull request merged**